src/pam_authy.c

```diff
@@ -0,0 +1,255 @@
+#define _GNU_SOURCE
+
+#include "config.h"
+#include <stdio.h>
+#include <unistd.h>
+#include <unistd.h>
+#include <string.h>
+#include <errno.h>
+#include <syslog.h>
+#include <jansson.h>
+#include <curl/curl.h>
+
+#include <security/pam_appl.h>
+#include <security/pam_modules.h>
+
+#include "pam_util.h"
+#include "pam_authy.h"
+
+typedef struct {
+       char *memory;
+       size_t size;
+} mblock_t;
+
+static size_t ctrl_curl_receive(void *content, size_t size, size_t nmemb,
+		void *user_mem)
+{
+       size_t realsize = size * nmemb;
+       mblock_t *mem = (mblock_t *)user_mem;
+
+       mem->memory = realloc(mem->memory, mem->size + realsize + 1);
+       if (mem->memory == NULL) {
+              return -ENOMEM;
+       }
+
+       memcpy(&(mem->memory[mem->size]), content, realsize);
+       mem->size += realsize;
+       mem->memory[mem->size] = 0;
+
+       return realsize;
+}
+
+static authy_rc_t authy_check_aproval(pam_handle_t *pamh, char *api_key, char *uuid)
+{
+	CURL *curl = NULL;
```

Comment:
Define variables later. E.g. instead of what's here now, do:

`CURL *curl = curl_easy_init();`

That way the scope of variables is reduced making it clearer where they are set.

Reply:
Agree, scope will be reduced